### PR TITLE
Document upsert JDBC sink primary key requirement

### LINKED ISSUE #1123: Document that upsert JDBC sinks must specify downstream primary key
### Related code PR

https://github.com/risingwavelabs/risingwave/pull/11042

### Which part(s) of the docs might be affected or should be updated? And how?

Document that upsert JDBC sinks must specify downstream primary key using the `primary_key` field

### Reference

_No response_

### PR DESCRIPTION
**Pull request by emile-00**
## Info

- **Description**
 Document upsert JDBC sink primary key requirement by updating PG sink and MySQL sink guides. Also update Sink Overview page to include AWS Kinesis and Delta Lake
  
- **Notes**

  [ Include any supplementary context or references here. ]

- **Related code PR**

  https://github.com/risingwavelabs/risingwave/pull/11042

- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/1123

## For reviewers

- **Preview**

  [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **upcoming** version of the documentation. ]

- **Key points**

  [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

**Comment by emile-00** [comment]
@StrikeW
I've edited the setup portion of the MySQL guide so that a primary key is defined when creating the table in MySQL. Please let me know if this is correct. 
Also, the Data delivery overview page has a section describing when a primary key needs to be defined for an upsert sink. Does this not apply to JDBC sinks? I'll make sure to update this section accordingly based on your reply.
Thank you!

File: docs/guides/sink-to-mysql.md [hunk 1]
```diff
@@ -145,7 +147,8 @@ All `WITH` options are required.
 |connector| Sink connector type must be `'jdbc'` for MySQL sink. |
 |jdbc.url| The JDBC URL of the destination database necessary for the driver to recognize and connect to the database.|
 |table.name| The table in the destination database you want to sink to.|
-|type|Data format. Allowed formats:<ul><li> `append-only`: Output data with insert operations.</li><li> `upsert`: Output data as a changelog stream. </li></ul> If creating an `upsert` sink, see the [Overview](/data-delivery.md) on when to define the primary key.|
+|type|Data format. Allowed formats:<ul><li> `append-only`: Output data with insert operations.</li></ul> `upsert`: Output data as a changelog stream. |
+|primary_key| Required if `type` is `upsert`. The primary key of the sink, which should match the primary key of the downstream table. |
```
Comment (StrikeW) [inline docs/guides/sink-to-mysql.md]: > The primary key of the sink

The semantic of `primary_key` is exactly the primary key of downstream table.

**Review by StrikeW — APPROVED** [review]
Rest lgtm

File: docs/data-delivery.md [hunk 2]
```diff
@@ -22,6 +22,16 @@ Currently, RisingWave supports the following sink connectors:
...
+
+- Delta Lake sink connector (`connector = 'deltalake'`)
+
+  With this connector, you can sink data from RisingWave to Delta Lake. For details about the syntax and parameters, see [Sink data to Delta Lake](/guides/sink-to-delta-lake.md).
+
+## Upsert sinks and primary keys
+
 For each sink, you can specify the data format. All sinks supports the `upsert` and `append-only` formats while Kafka also supports the `debezium` format. When creating an `upsert` sink, note whether or not you need to specify the primary key in the following situations.
 
 - If the downstream system supports primary keys and the table in the downstream system has a primary key, then RisingWave does not allow users to define a primary key when creating an upsert sink.
```
Comment (StrikeW) [inline docs/data-delivery.md]: This is outdated. => If the downstream system supports primary keys and the table in the downstream system has a primary key, user **must** specify the `primary_key` option when creating a **upsert JDBC sink**.

**Pull request merged**